[BUG] Странные падения при запуске анализа в Сонаре и через режим анализа командной строки

**Версия**
> Версия BSLLS либо плагина, который использует BSLLS
Проверял на разных - от 0.19.1 до сегодняшней ветки девелоп (коммит 0e4b3966fe08c8275a5e3afaced37fe332e0ecf1)
 - https://github.com/1c-syntax/bsl-language-server/commit/0e4b3966fe08c8275a5e3afaced37fe332e0ecf1

**Описание ошибки диагностики**
> Описание ошибки, замечания
Источник послужило падение в сонаре ночной сборки

- Описание ошибки https://pastebin.pl/view/e5bc9676
- выдается стабильно 
```
Diagnostic: Either [
  left = DeprecatedMethodCall
  right = null
]
```

посмотрел в коде, нашел подозрительное место
>похоже, я понял, где баг с падением расчета
- ![image](https://user-images.githubusercontent.com/2920817/137000804-ec3d6f5b-51db-4470-a6cf-5e12ccaf13d3.png)

Далее я запустил бсл лс в режиме анализа c отбором только по правилу `DeprecatedMethodCall`
и эта ошибка очень-очень много раз повторилась для кучи файлов на одном анализе (

например, на типовом файле - `src\cf\CommonModules\СтандартныеПодсистемыСервер\Ext\Module.bsl`

потом оказалось, что ошибка плавающая
- может появиться при первом запуске
- или нужно несколько раз запускать
- достаточно 2 или 3 раза для повторения

**Далее пошел искать ошибку на разных сборках**

- На 0.19.1 не падает
- На 0.20.ra1 падает
- на коммите 7f903c840be4e794eb166cea1c48f3f4444a9468 от референс индекса падает
- на коммите-родителе 9a4cefe791c27020cda0006f9b3a45340962f1f7 также падает, но с другой ошибкой
  - но уже с другой ошибкой, которую я также видел в своих испытаниях сегодня, но намного реже

```
Caused by: java.lang.NullPointerException: Cannot invoke "org.antlr.v4.runtime.Token.getLine()" because "endToken" is null
        at com.github._1c_syntax.bsl.languageserver.utils.Ranges.create(Ranges.java:85)
        at com.github._1c_syntax.bsl.languageserver.utils.Ranges.create(Ranges.java:75)
        at com.github._1c_syntax.bsl.languageserver.context.symbol.SymbolTree.getMethodSymbol(SymbolTree.java:121)
        at com.github._1c_syntax.bsl.languageserver.context.computer.CognitiveComplexityComputer.enterSub(CognitiveComplexityComputer.java:98)
        at com.github._1c_syntax.bsl.parser.BSLParser$SubContext.enterRule(BSLParser.java:2721)
        at org.antlr.v4.runtime.tree.ParseTreeWalker.enterRule(ParseTreeWalker.java:92)
        at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:42)
        at com.github._1c_syntax.bsl.languageserver.context.computer.CognitiveComplexityComputer.compute_aroundBody0(CognitiveComplexityComputer.java:85)
        at com.github._1c_syntax.bsl.languageserver.context.computer.CognitiveComplexityComputer$AjcClosure1.run(CognitiveComplexityComputer.java:1)
        at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:167)
        at com.github._1c_syntax.bsl.languageserver.aop.MeasuresAspect.measureComputers(MeasuresAspect.java:77)
        at com.github._1c_syntax.bsl.languageserver.context.computer.CognitiveComplexityComputer.compute(CognitiveComplexityComputer.java:78)
        at com.github._1c_syntax.bsl.languageserver.context.computer.CognitiveComplexityComputer.compute(CognitiveComplexityComputer.java:47)
        at com.github._1c_syntax.bsl.languageserver.context.DocumentContext.computeCognitiveComplexity(DocumentContext.java:339)

GitHub (https://github.com/1c-syntax/bsl-language-server/commit/9a4cefe791c27020cda0006f9b3a45340962f1f7)
Merge pull request #1857 from 1c-syntax/feature/complexity-lenses-thr… · 1c-syntax/bsl-language-server@9a4cefe
…eshold
```

**Пример кода**
> Пример кода (в блоке code или прикрепленный файл), на котором ошибка воспроизводится

Падало на разных файлах
например, на типовом файле - src\cf\CommonModules\СтандартныеПодсистемыСервер\Ext\Module.bsl

**Скриншоты**
> Скриншоты с подтверждением

**Дополнительная информация**
> Любая дополнительная информация, которая поможет в расследовании проблемы

--- a/src/main/java/com/github/_1c_syntax/bsl/languageserver/context/computer/MethodSymbolComputer.java
+++ b/src/main/java/com/github/_1c_syntax/bsl/languageserver/context/computer/MethodSymbolComputer.java
@@ -46,6 +46,7 @@ import javax.annotation.Nullable;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Optional;
 import java.util.Set;
@@ -60,7 +61,7 @@ public final class MethodSymbolComputer
     BSLParser.ANNOTATION_ATCLIENTATSERVER_SYMBOL);
 
   private final DocumentContext documentContext;
-  private final List<MethodSymbol> methods = new ArrayList<>();
+  private final Set<MethodSymbol> methods = new HashSet<>();
 
   public MethodSymbolComputer(DocumentContext documentContext) {
     this.documentContext = documentContext;

--- a/src/main/java/com/github/_1c_syntax/bsl/languageserver/context/computer/RegionSymbolComputer.java
+++ b/src/main/java/com/github/_1c_syntax/bsl/languageserver/context/computer/RegionSymbolComputer.java
@@ -33,7 +33,9 @@ import org.eclipse.lsp4j.Range;
 import java.util.ArrayDeque;
 import java.util.ArrayList;
 import java.util.Deque;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
 public final class RegionSymbolComputer
   extends BSLParserBaseVisitor<ParseTree>
@@ -42,7 +44,7 @@ public final class RegionSymbolComputer
   private final DocumentContext documentContext;
   private final Deque<Pair<RegionSymbol.RegionSymbolBuilder, BSLParser.RegionStartContext>> regionStack =
     new ArrayDeque<>();
-  private final List<RegionSymbol> regions = new ArrayList<>();
+  private final Set<RegionSymbol> regions = new HashSet<>();
 
   public RegionSymbolComputer(DocumentContext documentContext) {
     this.documentContext = documentContext;

--- a/src/main/java/com/github/_1c_syntax/bsl/languageserver/context/computer/VariableSymbolComputer.java
+++ b/src/main/java/com/github/_1c_syntax/bsl/languageserver/context/computer/VariableSymbolComputer.java
@@ -34,13 +34,15 @@ import org.antlr.v4.runtime.Token;
 import org.antlr.v4.runtime.tree.ParseTree;
 
 import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Optional;
+import java.util.Set;
 
 public class VariableSymbolComputer extends BSLParserBaseVisitor<ParseTree> implements Computer<List<VariableSymbol>> {
 
   private final DocumentContext documentContext;
-  private final List<VariableSymbol> variables = new ArrayList<>();
+  private final Set<VariableSymbol> variables = new HashSet<>();
 
   public VariableSymbolComputer(DocumentContext documentContext) {
     this.documentContext = documentContext;

